reconcile/dashdotdb_dvo.py

```diff
@@ -23,29 +23,48 @@ def __init__(self, dry_run, thread_pool_size):
         self.dashdotdb_url = secret_content['url']
         self.dashdotdb_user = secret_content['username']
         self.dashdotdb_pass = secret_content['password']
+        self.chunksize = secret_content.get('chunksize') or '20'
```

Comment:
I believe that the defaulkt should be `len(data)`, which means you don't set a default here...

Reply:
the chunksize is intended to be the max number of metrics we ship within a single post iteration, not the size the overall input. this seems to be working okay. i threw some debugs in and tried input of varying sizes (bigger and smaller than the limit). i'll make a small change in the chunk code, detailed below, to skip the chunking where we don't need it.

an example, showing where the last iteration of a list had a remainder of 2 result elements:
["data"]["result"][{metric{},value{}}]
```
{
	'status': 'success',
	'data': {
		'resultType': 'vector',
		'result': [{
			'metric': {
				'__name__': 'deployment_validation_operator_request_limit_validation',
				'endpoint': 'http-metrics',
				'exported_namespace': 'uhc-stage',
				'instance': '10.130.4.46:8383',
				'job': 'deployment-validation-operator-metrics',
				'kind': 'ReplicaSet',
				'name': 'uhc-acct-mngr-7fc4576947',
				'namespace': 'deployment-validation-operator',
				'pod': 'deployment-validation-operator-59b55df96c-xld4w',
				'service': 'deployment-validation-operator-metrics'
			},
			'value': [1611174950.067, '0']
		}, {
			'metric': {
				'__name__': 'deployment_validation_operator_request_limit_validation',
				'endpoint': 'http-metrics',
				'exported_namespace': 'visual-qontract-stage',
				'instance': '10.130.4.46:8383',
				'job': 'deployment-validation-operator-metrics',
				'kind': 'Deployment',
				'name': 'openshift-acme',
				'namespace': 'deployment-validation-operator',
				'pod': 'deployment-validation-operator-59b55df96c-xld4w',
				'service': 'deployment-validation-operator-metrics'
			},
			'value': [1611174950.067, '1']
		}]
	}
}
```

Reply:
..just to verify, i'm not mixing language syntax, this is valid:
```
j = {'one':'won', 'two':'too'}                                                  
print(f'j["one"]: {j.get("one") or "missing"}')  # j["one"]: won                
print(f'j["xyz"]: {j.get("xyz") or "missing"}')  # j["xyz"]: missing            
```

Reply:
I understand the concept... I'm just saying that 20 is a bad default, because it's not in the function signature, which makes it harder for the caller to know what the default is, and because a chunking mechanism should probably not chunk if the chunk size is not given.
As I said, no big deal.

Reply:
Totally fair. If I was adding something like that to the sre-toolbox, I'd opt for not stuffing a default in the init. Intent: wanted to be able to tune the chunksize at run-time, if we need, without waiting for the mammoth process of pushing a new qontract update. While we're still discovering the seams, in the context of the dvo class here it's probably okay as is. I did pre-emptively add a just-incase check if the property isn't set before we call the function here.